src/scheduler_helper.cc

```diff
@@ -0,0 +1,20 @@
+#include "../include/scheduler_helper.hh"
+
+void Scheduler_helper::set_thread_priority(pthread_t handle, std::string const &threadname) {
+
+    sched_param sch{0};
+    int policy = 0;
+    pthread_getschedparam(handle, &policy, &sch);
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+    sch.sched_priority = 20;
+    if (pthread_setschedparam(handle, SCHED_FIFO, &sch)) {
+        // NOLINTNEXTLINE(concurrency-mt-unsafe)
+        std::cerr << "Failed to set sched param: " << std::strerror(errno) << std::endl;
```

Comment:
This isn't a meaningful error message

Reply:
Can't think of any way to get more information when this is seperated though.

Reply:
Added the threadname to to the error output. 

I am aware that we will have arrays of some of the same threads, so would giving each thread a number be beneficial?

Reply:
I was thinking more like. "Failed to change priority for thread xxxx". The user doesn't care about sched params.

Reply:
that seems sensible to me.